### PR TITLE
feat: computed members added

### PR DESCRIPTION
**Pull request by waxb**
- chore: update changelog
- chore: update module from template and tests
- feat: computed members added

### CODE REVIEW

File: test/terramate_provider_versions.tm.hcl [hunk 1]
```diff
@@ -0,0 +1,5 @@
+globals {
+  provider                    = "google"
+  minimum_provider_version    = "4.0"
```
Comment (mariux) [inline test/terramate_provider_versions.tm.hcl:3]: please also drop support for <4.x in versions.tf

**Review by mariux — CHANGES_REQUESTED** [review]

**Review by mariux — CHANGES_REQUESTED** [review]

File: variables.tf [hunk 2]
```diff
@@ -19,8 +19,19 @@ variable "members" {
...
 
   validation {
-    condition     = alltrue([for m in var.members : can(regex("^(allUsers|allAuthenticatedUsers|(user|serviceAccount|group|domain|principalSet):)", m))])
-    error_message = "The value must be a non-empty list of strings where each entry is a valid principal type identified with `allUsers`, `allAuthenticatedUsers` or prefixed with `user:`, `serviceAccount:`, `group:`, `domain:` or `principalSet:`."
+    condition     = alltrue([for m in var.members : can(regex("^(allUsers|allAuthenticatedUsers|(user|serviceAccount|group|domain|computed):)", m))])
+    error_message = "The value must be a non-empty list of strings where each entry is a valid principal type identified with `allUsers`, `allAuthenticatedUsers` or prefixed with `user:`, `serviceAccount:`, `group:`, `domain:` or `computed`."
+  }
+}
+
+variable "computed_members_map" {
```
[the remark this answers is not in the document]
Comment (waxb) [reply variables.tf:27]: i copied this a long time ago, my bad!

File: variables.tf [hunk 3]
```diff
@@ -19,8 +19,19 @@ variable "members" {
   default     = []
 
   validation {
-    condition     = alltrue([for m in var.members : can(regex("^(allUsers|allAuthenticatedUsers|(user|serviceAccount|group|domain|principalSet):)", m))])
-    error_message = "The value must be a non-empty list of strings where each entry is a valid principal type identified with `allUsers`, `allAuthenticatedUsers` or prefixed with `user:`, `serviceAccount:`, `group:`, `domain:` or `principalSet:`."
+    condition     = alltrue([for m in var.members : can(regex("^(allUsers|allAuthenticatedUsers|(user|serviceAccount|group|domain|computed):)", m))])
```
Comment (mariux) [inline variables.tf:22]: :boom: this disables OIDC ;)

**Review by mariux — CHANGES_REQUESTED** [review]

**Review by zied-elouaer — DISMISSED** [review]
LGTM

File: variables.tf [hunk 4]
```diff
@@ -19,8 +19,19 @@ variable "members" {
...
+  }
+}
+
+variable "computed_members_map" {
+  type        = map(string)
+  description = "(Optional) A map of members to replace in 'var.members' or in members of 'policy_bindings' to handle terraform computed values."
+  default     = {}
+
+  validation {
+    condition     = alltrue([for k, v in var.computed_members_map : can(regex("^(allUsers|allAuthenticatedUsers|(user|serviceAccount|group|domain):)", v))])
```
Comment (mariux) [inline variables.tf:33]: principalSet is missing.

**Review by mariux — CHANGES_REQUESTED** [review]

**Review by mariux — APPROVED** [review]

**Pull request merged**